packages/l2b/src/implementations/discovery-ui/main.ts

```diff
@@ -46,6 +49,25 @@ export function runDiscoveryUi() {
     res.sendFile(join(STATIC_ROOT, 'index.html'))
   })
 
+  // Start executing one of predefined commands
+  // and stream the output back to the client
+  app.post('/api/terminal/execute', (req, res) => {
+    const { command, project, chain } = req.body
+    if (!command || !project || !chain) {
+      res.status(400).send('Missing required parameters')
+      return
+    }
+    if (command !== 'discover') {
+      res.status(400).send('Invalid command')
+    }
+
+    res.setHeader('Content-Type', 'text/plain')
+    executeTerminalCommand(
```

Comment:
I was thinking we could utilize https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events for this. However I don't want it getting too complex, maybe this is actually enough for what we need, we just need a better cancellation policy.

Reply:
Reimplemented